cl/free_look_email/app.py

```diff
@@ -0,0 +1,149 @@
+    original from: http://wiki.python.org/moin/PythonDecoratorLibrary#Retry
+    :param ExceptionToCheck: the exception to check. may be a tuple of
+    exceptions to check
+    :type ExceptionToCheck: Exception or tuple
+    :param tries: number of times to try (not retry) before giving up
+    :type tries: int
+    :param delay: initial delay between retries in seconds
+    :type delay: int
+    :param backoff: backoff multiplier e.g. value of 2 will double the delay
+    each retry
+    :type backoff: int
+    """
+
+    def deco_retry(f):
+        @wraps(f)
+        def f_retry(*args, **kwargs):
+            mtries, mdelay = tries, delay
+            while mtries > 1:
+                try:
+                    return f(*args, **kwargs)
+                except ExceptionToCheck as e:
+                    msg = "%s, Retrying in %d seconds..." % (str(e), mdelay)
+                    print(msg)
+                    time.sleep(mdelay)
+                    mtries -= 1
+                    mdelay *= backoff
+            return f(*args, **kwargs)
+
+        return f_retry  # true decorator
+
+    return deco_retry
+
+
+def get_email_record_from_event(event):
+    return decamelize(event["Records"][0]["ses"]["mail"])
+
+
+def get_email_receipt_from_event(event):
+    return decamelize(event["Records"][0]["ses"]["receipt"])
+
+
+def get_spam_verdict(receipt):
+    return receipt["spam_verdict"]["status"]
+
+
+def get_virus_verdict(receipt):
+    return receipt["virus_verdict"]["status"]
+
+
+def get_spf_verdict(receipt):
+    return receipt["spf_verdict"]["status"]
+
+
+def get_dkim_verdict(receipt):
+    return receipt["dkim_verdict"]["status"]
+
+
+def get_dmarc_verdict(receipt):
+    return receipt["dmarc_verdict"]["status"]
+
+
+def get_combined_log_message(email):
+    subject = None
+    for header in email["headers"]:
+        if header["name"] == "Subject":
+            subject = header["value"]
+
+    return "Email with {subject}, from {source} to {destination}".format(
+        subject=subject,
+        source=email["source"],
+        destination=email["destination"],
+    )
+
+
+def validation_failure(email, receipt, verdict):
+    print(
+        "{combined} failed with spam verdict {verdict}".format(
+            combined=get_combined_log_message(email), verdict=verdict
+        )
+    )
+    return {
+        "statusCode": 424,
+        "body": json.dumps(receipt),
+    }
+
+
+@retry(
+    (
+        ConnectionError,
+        HTTPError,
+        Timeout,
+    ),
+    tries=10,
+    delay=5,
+    backoff=3,
+)
+def send_to_court_listener(email, receipt):
+    # TODO - Docker internal host here should be an environment variable.
+    print(
+        "{combined} sending to Court Listener API.".format(
+            combined=get_combined_log_message(email)
+        )
+    )
+    court_listener_response = requests.post(
+        "http://host.docker.internal:8000/api/rest/v3/free-look-email/",
```

Comment:
OK, I think I get it now. You have an API that the tests run against, and that uses AWS's API service, but in the real world we'll use a env to send things to CL instead? That makes sense. Bummer that you need a whole AWS service for this though (seems like a lot, since I've never used their API service). 

Re this line, can I suggest a slightly different URL:

```suggestion
        "http://host.docker.internal:8000/api/rest/v3/recap-email/",
```

That'll be more consistent with the other API URLs: 

 - `recap`: Where the extension sends things
 - `recap-fetch`: Where folks can fetch PACER content
 - `recap-query`: Where the extension queries for documents
 - `recap-email`: Where emails get posted

Reply:
We are now pulling this from an environment variable, so the naming is whatever we want to do on the cl side.